mlflow/pipelines/step.py

```diff
@@ -164,6 +165,15 @@ def _run(self, output_directory: str) -> BaseCard:
         """
         pass
 
+    @experimental
+    @abc.abstractmethod
+    def _init_from_step_config(self) -> None:
+        """
+        This function is responsible for validating the step config for
```

Comment:
The function is also used to load the parsed step config. Can we include that in the docstring? I think that `_validate_and_apply_step_config` would be a clearer method name.